internal/leakcheck/leakcheck.go

```diff
@@ -298,3 +285,123 @@ func NewLeakChecker(logger Logger) *LeakChecker {
 	SetTrackingBufferPool(logger)
 	return &LeakChecker{logger: logger}
 }
+
+type timerFactory struct {
+	logger Logger
+
+	originalTimeAfterFunc func(time.Duration, func()) internal.Timer
```

Comment:
Is there any need for this?  As opposed to just directly calling `time.AfterFunc`